controllers/flowcollector_controller.go

```diff
@@ -101,6 +113,40 @@ func (r *FlowCollectorReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	clientHelper := r.newClientHelper(desired)
+
+	// reconcile operator content only when CRD / Owned object has changed
+	// this avoid watched objects like secret to call unecessary reconcile steps
```

Comment:
May be we should start to have multiple goroutine.
Each one watching and reconciling independent part.
So we would avoid this kind of check.

Reply:
This would help to split the code and avoid checking `req.Name` all the time, however, how can we manage status & conditions with this approach ?

Let's say we have two controllers, `flowcollector_controller` (for our components) & `dependencies_controller` (for secrets & dependent operators). 
If a secret or a dependent operator is pending on `dependencies_controller,` we should not mark status as `Ready` from `flowcollector_controller`.

Does this mean we should store some states from each controller in `FlowCollectorStatus` as same as Namespace and rely on these to set the final conditions ?